File: DVLD_Project/Form20.vb
Imports BussnessLayerForDVLD
Imports DVLD_Project.Properties
Imports System
Imports System.Windows.Forms

Namespace DVLD_Project
    Public Partial Class Form20
        Inherits Form
        Private _Fees As Single = 10.0F
        Private _ExamMode As Integer = 0, _DL As Integer = 0, _APPID As Integer = 0
        Private _Mode As String = ""
        Public Sub New()
            InitializeComponent()
        End Sub

        Public Sub New(Mode As String, ExamMode As Integer, DL As Integer, TestID As Integer, ClassID As Integer)
            InitializeComponent()
            _Mode = Mode
            _ExamMode = ExamMode
            _DL = DL
            testMainInformations1.LoadControal(Mode, ExamMode, DL, TestID, ClassID)
            'testMainInformations1.LoadControal(Mode,DL,TestID,ClassID);
            If ExamMode = 2 Then
                pictureBox1.Image = Resources.icons8_compose_641
                'label1.Text = "Written Test";
                _Fees = 20.0F
            ElseIf ExamMode = 3 Then
                pictureBox1.Image = Resources.icons8_car_64
                ' label1.Text = "Driving Test";
                _Fees = 30.0F
            End If
            Dim t As Integer = testMainInformations1.GetTotalTryes()
            If t = 0 OrElse Equals(_Mode, "EDITE") Then
                reTakeExam1.Enabled = False
            Else

                reTakeExam1.Enabled = True
            End If
            reTakeExam1.LoadControl(t, _Fees)
        End Sub


        Public Sub loadAppID(AppID As Integer)
            _APPID = AppID

        End Sub

        Private Sub Form20_Load(sender As Object, e As EventArgs)

        End Sub

        Private Sub testMainInformations1_Load(sender As Object, e As EventArgs)

        End Sub

        Private Sub guna2Button1_Click(sender As Object, e As EventArgs)
            If Equals(_Mode, "Add") Then
                Dim _TestA As Integer = clsBussnesLayer.AddTestA(0, _ExamMode, _DL, Date.Now, reTakeExam1.GetTotalFees(), clsBussnesLayer.GetUserID(), False)
            Else
                Dim _TestA As Integer = clsBussnesLayer.UpDateTestAPPDate(_APPID, testMainInformations1.GetNewDateTime())
            End If

        End Sub
    End Class
End Namespace

File: DVLD_Project/Form20.cs
using BussnessLayerForDVLD;
using DVLD_Project.Properties;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace DVLD_Project
{
    public partial class Form20 : Form
    {
        float _Fees = 10.0f; int _ExamMode = 0, _DL = 0, _APPID = 0;
        string _Mode = "";
        public Form20()
        {
            InitializeComponent();
        }

        public Form20(string Mode, int ExamMode, int DL, int TestID, int ClassID)
        {
            InitializeComponent();
            _Mode = Mode;
            _ExamMode = ExamMode;
            _DL = DL;
            testMainInformations1.LoadControal(Mode, ExamMode, DL, TestID, ClassID);
            //testMainInformations1.LoadControal(Mode,DL,TestID,ClassID);
            if (ExamMode == 2)
            {
                pictureBox1.Image = Resources.icons8_compose_641;
                _Fees = 20.0f;
                //label1.Text = "Written Test";
            }
            else if (ExamMode == 3)
            {
                pictureBox1.Image = Resources.icons8_car_64;
                // label1.Text = "Driving Test";
                _Fees = 30.0f;
            }
            int t = testMainInformations1.GetTotalTryes();
            if (t == 0 || _Mode == "EDITE")
            {
                reTakeExam1.Enabled = false;
            }
            else
            {

                reTakeExam1.Enabled = true;
            }
            reTakeExam1.LoadControl(t, _Fees);
        }


        public void loadAppID(int AppID)
        {
            _APPID = AppID;

        }

        private void Form20_Load(object sender, EventArgs e)
        {

        }

        private void testMainInformations1_Load(object sender, EventArgs e)
        {

        }

        private void guna2Button1_Click(object sender, EventArgs e)
        {
            if (_Mode == "Add")
            {
                int _TestA = clsBussnesLayer.AddTestA(0, _ExamMode, _DL, DateTime.Now, reTakeExam1.GetTotalFees(), clsBussnesLayer.GetUserID(), false);
            }
            else
            {
                int _TestA = clsBussnesLayer.UpDateTestAPPDate(_APPID, testMainInformations1.GetNewDateTime());
            }

        }
    }
}
